src/components/ReconnectPrompt.svelte

```diff
@@ -32,17 +23,102 @@
     startConnectionProcess();
     stopOfferingReconnect();
   };
+
+  const content = {
+    heading: '',
+    subtitle: '',
+    listHeading: '',
+    bulletOne: '',
+    bulletTwo: '',
+  };
+
+  const textIdPrefix =
+    trigger === 'automatic' ? 'disconnectedWarning' : 'reconnectFailed';
+
+  switch (type) {
```

Comment:
My initial reaction is to query whether a version of this that didn't mutate would be easier to understand. It seems like it would be no longer and be easier to read? The generic case would need to populate the blanks but that happens in the default scenario anyway (where all other cases override). May well have missed something, no worries if you tried it and this is better.

Reply:
[1f1dd77](https://github.com/microbit-foundation/ml-trainer/pull/206/commits/1f1dd7778334800b998e54ce428b75f987f7abfe)